engines/pegasus/metaengine.cpp

```diff
@@ -78,6 +78,14 @@ class PegasusMetaEngine : public AdvancedMetaEngine {
 	int getMaximumSaveSlot() const override { return 999; }
 	void removeSaveState(const char *target, int slot) const override;
 	Common::KeymapArray initKeymaps(const char *target) const override;
+	Common::String getSavegameFile(int saveGameIdx, const char *target) const override {
+		if (saveGameIdx == kSavegameFilePattern)
+			return Common::String::format("pegasus-*.sav");
+		Common::StringArray fileNames = Pegasus::PegasusEngine::listSaveFiles();
+		if (saveGameIdx < fileNames.size())
+			return fileNames[saveGameIdx];
+		return AdvancedMetaEngine::getSavegameFile(saveGameIdx, target);
```

Comment:
For non-existent save files, this would return a name that does not follow the expected "pegasus-*.sav" pattern. Is that a good idea? Maybe it would be better to return an unused name (one that is not in the `fileNames` array), and follows the pattern? If the function is going to be used to name a save file (e.g. when we copy one from one slot to another), it could be an issue if the name does not follow the pattern as it will then not be listed the next time we call `listSaveFiles()`. Even better might be to make sure it is after the last element of `fileNames` alphabetically, but that might get complicated.

But that might be all hypothetical if the only case where we use this function with an inexistent slot is to copy a normal savegame from the autosave slot to another slot, as this should never happen with this engine (unless the user happens to create a normal savegame and name it `AutoSave`, but that would be sneaky).

Reply:
I doubt this code path can be reached. Handled anyway :)